### ISSUE TITLE
You realize StreamStep and ParseStep are the same datatype, right?

### ISSUE THREAD

**Issue by 0b1kn00b** [haxetink/tink_streams#2]
```
enum ParseStep<Result> {
  Failed(e:Error);
  Done(r:Result);
  Progressed;
}


enum StreamStep<T> {
  Data(data:T);
  End;
  Fail(e:Error);
}
```

**Comment by 0b1kn00b** [comment]
[general Chunk datatype](https://github.com/0b1kn00b/stx_chunk/blob/development/src/main/haxe/stx/Chunk.hx)

**Comment by back2dos** [comment]
They are quite different. A stream progresses with data and ends with nothing, while a parser progresses with nothing and ends with data. 

Generally I found that streams of bytes and streams of actual data should not be treated the same, which is why I consider `tink.streams.Stream` and `tink.io.Source` to be fundamentally different. One could express the latter through the former, but my attempts to do that resulted in abysmal performance. I also found the generalization yielded little practical value. IO streams need to be parsed, chunked or piped. Data streams need to be filtered, mapped, aggregated, merged etc.
